### PR TITLE
perf: Collision detection improved

### PR DESCRIPTION
**Pull request by OriolSors**
Improved the criteria for discarding rigidbodies before the contact test.

Big FPS improvement.

### CODE REVIEW

File: Source/Modules/ModulePhysics.cpp [hunk 1]
```diff
@@ -126,45 +140,49 @@ struct ContactResultCallback : public btCollisionWorld::ContactResultCallback
...
+		}
+
 		ContactResultCallback result;
+
 		dynamicsWorld->contactTest(obj, result);
 
 		if (result.collisionDetected)
 		{
-			if (obj->getUserPointer() != nullptr)
+			for (int j = 0; j < result.othersRigidBody.size(); j++)
```
Comment (markmeliaa) [inline Source/Modules/ModulePhysics.cpp:165]: `for_each` loops are considered better than standard `for` loops IIRC.

**Review by AlexisDieu — APPROVED** [review]
I tested and profile it and working perfectly 0ms to 1 ms depending of the quadtree detection ! Good job !

**Review by markmeliaa — APPROVED** [review]

**Review by totisama — APPROVED** [review]
Good job
It works fine!

**Review by MacXxs — APPROVED** [review]
Is this a dream? 😭

**Pull request merged**